src/main/java/teammates/ui/webapi/action/PutCourseStudentDetailsEditAction.java

```diff
@@ -66,8 +66,8 @@ public ActionResult execute() throws EntityNotFoundException {
         student.comments = SanitizationHelper.sanitizeTextField(student.comments);
 
         try {
+            student.nullToEmptyString();
             StudentAttributes originalStudentAttribute = logic.getStudentForEmail(courseId, studentEmail);
-            student.updateWithExistingRecord(originalStudentAttribute);
 
             boolean isSectionChanged = student.isSectionChanged(originalStudentAttribute);
```

Comment:
Can we assume that `student` has full attributes `name`, `email`, `team`, `section`, `comments` at this point? If yes, then this line and the line below can function correctly.

Reply:
@xpdavid Yes, `name`, `email`, `team`, `section`, `comments` would have values before the try block is executed, `null` if values are empty. (`getRequestParamValue` grabs the value of the constant and if they are not present, returns `null`)

The newly added function `student.nullToEmptyString();` converts all `null` values except `team` and `section` to empty strings. 

This line and the line below would also check if they are null and not equals to original student's section/team and then compute the boolean result accordingly.

Reply:
- If we can assume that they are `null`, why not using `getNotNullRequestParamValue()` so there is no need to have `nullToEmptyString()`

- Even if we need `nullToEmptyString()`, I guess this is the only place where this method is used. Why not inline it in the action.

- This raises another question: why the frontend give `null` input?

Reply:
If the user provides an empty input, it is processed as `null` after `getRequestParamValue`. 

The fields `section` and `comments` can be allowed empty.

Reply:
Then the problem is "Why the frontend pass null instead of empty string"?

Reply:
Good point, I found out why this behavior is seen. This [line](https://github.com/TEAMMATES/teammates/blob/27a092a602e6aacf698a225a8617684aaed6f0d7/src/web/services/http-request.service.ts#L27) would ignore the param if it is an empty string as `paramsMap[key]` evaluated to false.

@wkurniawan07 Is it cool with you if I change to this line to `if (paramsMap[key] || paramsMap[key] === '')` ?

Reply:
> Is it cool with you if I change to this line to `if (paramsMap[key] || paramsMap[key] === '')` ?

@tanhengyeow Pretty sure that is going to break some stuffs.

Reply:
Also, can I advice to wait until #9488 is merged? The problem might very well be non-existent by then.